src/server/controllers/webhook-events/github-app-install.ts

```diff
@@ -0,0 +1,72 @@
+import * as fse from 'fs-extra';
+import gql from 'graphql-tag';
+import * as path from 'path';
+
+import {RepoIdQuery} from '../../../types/gql-types';
+import {github} from '../../../utils/github';
+import {WebHookHandleResponse} from '../../apis/github-webhook';
+import {githubAppModel} from '../../models/githubAppModel';
+
+export interface InstallHook {
+  action: 'created';
+  installation: {
+    id: number; repository_selection: 'all' | 'selected';
+    permissions: {[name: string]: string;};
+    events: string[];
+    account: {login: string; avatar_url: string; type: 'User' | 'Organization';}
+    repositories: Array<{id: number; name: string; full_name: string}>
+  };
+}
+
+export async function handleGithubAppInstall(hookBody: InstallHook):
+    Promise<WebHookHandleResponse> {
+  await githubAppModel.addInstallation({
+    installationId: hookBody.installation.id,
+    permissions: hookBody.installation.permissions,
+    events: hookBody.installation.events,
+    repository_selection: hookBody.installation.repository_selection,
+    type: hookBody.installation.account.type,
+    login: hookBody.installation.account.login,
+    avatar_url: hookBody.installation.account.avatar_url,
+  });
+
+  const tokens = await fse.readJson(
+      path.join(__dirname, '..', '..', '..', '..', 'tokens.json'));
+
+  for (const repo of hookBody.installation.repositories) {
```

Comment:
Does this return the full list if `all` is selected?

Reply:
yes it does